src/moscot/problems/time/_mixins.py

```diff
@@ -427,102 +426,72 @@ def pull(
     @property
     def prior_growth_rates(self: TemporalMixinProtocol[K, B]) -> Optional[pd.DataFrame]:
         """Return the prior estimate of growth rates of the cells in the source distribution."""
-        # TODO(michalk8): FIXME
         cols = ["prior_growth_rates"]
         df_list = [
-            pd.DataFrame(problem.prior_growth_rates, index=problem.adata.obs.index, columns=cols)
+            pd.DataFrame(problem.prior_growth_rates, index=problem.adata.obs_names, columns=cols)
             for problem in self.problems.values()
         ]
-        tup = list(self)[-1]
-        df_list.append(
-            pd.DataFrame(
-                np.full(
-                    shape=(len(self.problems[tup].adata_tgt.obs), 1),
-                    fill_value=np.nan,
-                ),
-                index=self.problems[tup].adata_tgt.obs.index,
-                columns=cols,
-            )
-        )
-        return pd.concat(df_list, verify_integrity=True)
+        df_1 = pd.concat(df_list, verify_integrity=True)
+        indices_remaining = set(self.adata.obs_names) - set(df_1.index)
+        df_2 = pd.DataFrame(np.nan, index=list(indices_remaining), columns=cols)
+
+        return pd.concat([df_1, df_2], verify_integrity=True)
 
     @property
     def posterior_growth_rates(self: TemporalMixinProtocol[K, B]) -> Optional[pd.DataFrame]:
         """Return the posterior estimate of growth rates of the cells in the source distribution."""
-        # TODO(michalk8): FIXME
         cols = ["posterior_growth_rates"]
         df_list = [
-            pd.DataFrame(problem.posterior_growth_rates, index=problem.adata.obs.index, columns=cols)
+            pd.DataFrame(problem.posterior_growth_rates, index=problem.adata.obs_names, columns=cols)
```

Comment:
Does this work when `posterior_growth_rates` are `None`?

Reply:
Yes, it returns a `pd.DataFrame` with `NaN` entries. But I think you're right, let's just return `None` then.